src/Service/UpdateWithParamsTrait.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace ChartMogul\Service;
+
+use ChartMogul\Http\ClientInterface;
+
+/**
+* @codeCoverageIgnore
+*/
+trait UpdateWithParamsTrait
+{
+    public static function update(array $id = [], array $data = [], ClientInterface $client = null)
+    {
+        return (new RequestService($client))
+            ->setResourceClass(static::class)
+            ->setResourcePath(static::RESOURCE_PATH)
+            ->update($id, $data);
```

Comment:
`$id` should be a part of `$data` and we have to support `external_id` and `data_source_uuid` as an alternative to `$id`